microsoft-edge/webview2/concepts/print.md

```diff
@@ -0,0 +1,736 @@
+  // Return the user selected printer name.
+}
+
+// Function to get print settings for the selected printer.
+// You may also choose get the capabilties from the native printer API, display to the user to get
+// the print settings for the current web page and for the selected printer.
+CoreWebView2PrintSettings GetSelectedPrinterPrintSettings(string printerName)
+{
+  CoreWebView2PrintSettings printSettings = null;
+  printSettings = WebViewEnvironment.CreatePrintSettings();
+  printSettings.ShouldPrintBackgrounds = true;
+  printSettings.ShouldPrintHeaderAndFooter = true;
+
+  return printSettings;
+
+  // or
+  // Get PrintQueue for the selected printer and use GetPrintCapabilities() of PrintQueue from System.Printing
+  // to get the capabilities of the selected printer.
+  // Display the printer capabilities to the user along with the page settings.
+  // Return the user selected settings.
+}
+```
+
+**APIs:**
+
+* [CoreWebView2.PrintAsync() Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.printasync#microsoft-web-webview2-core-corewebview2-printasync)
+* [CoreWebView2PrintSettings Class](/dotnet/api/microsoft.web.webview2.core.corewebview2printsettings)
+
+<!-- ------------------------------ -->
+
+# [C++](#tab/cpp)
+
+```cpp
+struct SamplePrintSettings
+{
+  COREWEBVIEW2_PRINT_ORIENTATION Orientation = COREWEBVIEW2_PRINT_ORIENTATION_PORTRAIT;
+  int Copies = 1;
+  int PagesPerSide = 1;
+  std::wstring Pages = L"";
+  COREWEBVIEW2_PRINT_COLLATION Collation = COREWEBVIEW2_PRINT_COLLATION_DEFAULT;
+  COREWEBVIEW2_PRINT_COLOR_MODE ColorMode = COREWEBVIEW2_PRINT_COLOR_MODE_DEFAULT;
+  COREWEBVIEW2_PRINT_DUPLEX Duplex = COREWEBVIEW2_PRINT_DUPLEX_DEFAULT;
+  COREWEBVIEW2_PRINT_MEDIA_SIZE Media = COREWEBVIEW2_PRINT_MEDIA_SIZE_DEFAULT;
+  double PaperWidth = 8.5;
+  double PaperHeight = 11;
+  double ScaleFactor = 1.0;
+  bool PrintBackgrounds = false;
+  bool HeaderAndFooter = false;
+  bool ShouldPrintSelectionOnly = false;
+  std::wstring HeaderTitle = L"";
+  std::wstring FooterUri = L"";
+};
+
+// Function to get printer name by displaying installed printers list to the user and
+// return user selected printer.
+std::wstring AppWindow::GetPrinterName()
+{
+  // Use win32 EnumPrinters function to get locally installed printers.
+  // Display the printer list to the user and get the user desired printer to print.
+  // Return the user selected printer name.
+}
+
+// Function to get print settings for the selected printer.
+// You may also choose get the capabilties from the native printer API, display to the user to get
+// the print settings for the current web page and for the selected printer.
+SamplePrintSettings AppWindow::GetSelectedPrinterPrintSettings(std::wstring printerName)
+{
+  SamplePrintSettings samplePrintSettings;
+  samplePrintSettings.PrintBackgrounds = true;
+  samplePrintSettings.HeaderAndFooter = true;
+
+  return samplePrintSettings;
+
+  // or
+  //
```

Comment:
I'm not clear on the "Or" construction. Could be clearer.  like "Or, as an alternative to the above function/ code block:"
```suggestion
```